compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -4212,6 +4213,68 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
           typedExpr(cmp, defn.BooleanType)
       case _ =>
 
+  // if expression in statement position (of template or block)
+  // looks like a useful value that should not be ignored, warn and return true
+  // User specifies that an expression is boring by ascribing `e: Unit`.
+  // The subtree `e` will bear an attachment, but may be wrapped in adaptations.
+  private def checkInterestingResultInStatement(t: Tree)(using Context): Boolean = {
+    def isUninterestingSymbol(sym: Symbol): Boolean =
+      sym == NoSymbol ||
+      sym.isConstructor ||
+      sym.is(Package) ||
+      sym.isPackageObject ||
+      sym.isClass ||
```

Comment:
The ported logic in scala 2 has `sym.isPackageObjectOrClass`, which is defined as
```scala
def isPackageObjectOrClass = isPackageObject || isPackageObjectClass
```
however `isClass` and `isPackageObjectClass` don't seem equivalent

Reply:
Right, must've missed this. Thanks